src/workers/parser/getScopes/visitor.js

```diff
@@ -576,25 +572,31 @@ const scopeCollectionVisitor = {
         }
       });
     } else if (t.isIdentifier(node) && t.isReferenced(node, parentNode)) {
-      const identScope = findIdentifierInScopes(state.scope, node.name);
-      if (identScope) {
-        identScope.bindings[node.name].refs.push({
-          type: "ref",
-          start: fromBabelLocation(node.loc.start, state.sourceId),
-          end: fromBabelLocation(node.loc.end, state.sourceId),
-          meta: buildMetaBindings(state.sourceId, node, ancestors)
-        });
+      let freeVariables = state.freeVariables.get(node.name);
```

Comment:
what's this doing? can you add some comments here for what a free variable is

Reply:
Totally.